huxley/core/models.py

```diff
@@ -372,10 +374,12 @@ def country(self):
     def committee(self):
         return self.assignment.committee
 
-    @property
-    def school(self):
-        return self.assignment.school
+    def save(self, *args, **kwargs):
+        if (self.assignment and self.school and self.school.id != self.assignment.school.id):
```

Comment:
Models also have `_id` fields for their foreign keys so that you can compare keys without having to load the objects themselves. So you can write this as follows:
```python
if self.assignment_id and self.school_id and self.school_id != self.assignment.school_id:
    raise ValidationError(...)
```
This saves us from loading at max two school objects.